### PR TITLE
Handle '[!] (plugin esbuild) Error: Invalid option: "jsxFactory"' error

### PR DESCRIPTION
**Pull request by jdalrymple**
The latest version of esbuild throws an error when encountering undefined config properties. This code removes undefined properties before sending the config to esbuild

### CODE REVIEW

**Comment by jdalrymple** [comment]
Are there any steps I should take to hasten this PR? :D

File: src/index.ts [hunk 1]
```diff
@@ -125,14 +125,18 @@ export default (options: Options = {}): Plugin => {
...
-      const result = await service.transform(code, {
+      const config = {
         loader,
         target: options.target || defaultOptions.target || 'es2017',
         jsxFactory: options.jsxFactory || defaultOptions.jsxFactory,
         jsxFragment: options.jsxFragment || defaultOptions.jsxFragment,
         define: options.define,
         sourcemap: options.sourceMap,
-      })
+      } as Record<string, unknown>
```
Comment (egoist) [inline src/index.ts:135]: can you change it to the proper type instead of `Record`?

Comment (jdalrymple) [reply src/index.ts]: fixed!

Comment (egoist) [reply src/index.ts]: can you resolve the conflict?

**Comment by jdalrymple** [comment]
resolved

**Pull request closed without merging**